README.md

```diff
@@ -2,135 +2,75 @@
-
-```diff
-@prefix hydra: <http://www.w3.org/ns/hydra/core#> .
-+@prefix code: <https://code.declared.at/> .
-+@prefix hydra-box: <http://hydra-box.org/schema/> .
-
-    hydra:title "Get the entrypoint resource" ;
--    hydra:method "GET"
-+    hydra:method "GET" ;
-+    code:implementedBy [
-+      a hydra-box:SparqlQuery ;
-+      code:link <file:hydra/entrypoint.get.sparql>
-+    ]
+```javascript
+const api = await Api.fromFile('api.ttl', {
+  path: '/api',
+  codePath: __dirname
+})
 ```
 
-The simplest possible content of the `hydra/entrypoint.get.sparql` file is a `DESCRIBE <https://example.app/>` query.
-
-Fianally, an explicit declaration of the type of the entrypoint resource is necessary so that hydra-box can set up 
-the express router.
+Once both objects are created, the middleware can be used:
 
-```diff
-+<> a <api#Entrypoint> .
+```javascript
+const app = express()
+app.use(hydraBox(api, store))
+app.listen(9000)
 ```
 
-### Create an express application using hydra-box
-
-With the first operation and its implementation declared, it's time to create a minimal express app in `server.js` file.
-
-```js
-const path = require('path')
-const express = require('express')
-const hydraBox = require('hydra-box')
-
-function hydraMiddleware () {
-  return hydraBox.fromUrl('/api', 'file://' + path.join(__dirname, 'hydra/api.ttl'), {
-    sparqlEndpointUrl: 'http://example.app/sparql',
-    contextHeader: '/context/'
-  })
-}
-
-Promise.resolve()
-  .then(async () => {
-    const app = express()
-    app.use(await hydraMiddleware())
-    app.listen(9090)
-  })
-  .catch(err => console.error(err))
-```
-
-Few things to notice here:
-
-* The `hydraBox.fromUrl` method takes a `file://` URL and not a filesystem path. It means that also a remote graph can be used
-* The aforementioned method is async, hence the `Promise.resolve()` wrapping the bootstrap
-* Its first argument is the desired path to the API Documentation resource. It must match the one used earlier
-* Not shown here, `cors` is immediately necessary to expose `Link` headers to browser clients outside your domain
-
-## Releasing
-
-```shell
-npm run release
-git push --follow-tags origin linting && npm publish
+### Operation
+
+The operations must implement a [Express routing handler](http://expressjs.com/en/starter/basic-routing.html) interface (`(req, res, next) => {}`).
+Hydra-Box adds the [@rdfjs/express-handler](https://github.com/rdfjs-base/express-handler) to handle incoming and outgoing RDF data.
+For `GET` requests with a matching IRI Template, the `.dataset()` and `.quadStream()` as defined by `express-handler` are also available to read the given variables.
+Additionally there is a `hydra` property assigned to `req` that contains more data about the request: 
+
+```javascript
+  req.hydra = {
+    // Api object given as argument to the middleware
+    api,
+ 
+    // RDF/JS Store object given as argument to the middleware
+    store,
```

Comment:
I just figured that maybe the name `store` is a bit technical and not descriptive.

Common term in API space is **state** or **resource state**. So maybe something like `stateStore`? Or `resourceStateStore` but rather long...